processor/src/main/resources/org/mapstruct/ap/internal/model/NestedPropertyMappingMethod.ftl

```diff
@@ -21,7 +21,11 @@
     <#if !entry.presenceChecker?? >
     <#if !entry.type.primitive>
     if ( ${entry.name} == null ) {
-        return ${returnType.null};
+        <#if returnType.isOptionalType()>
+            return Optional.empty();
```

Comment:
Should probably use `InitDefaultValue` here instead of hardcoding `Optional.empty()`, but this ModelElement is missing the required `factoryMethod`.